src/libraries/Native/Unix/System.Native/pal_networking.c

```diff
@@ -519,6 +503,152 @@ int32_t SystemNative_GetHostEntryForName(const uint8_t* address, int32_t address
     return ret;
 }
 
+#if HAVE_GETADDRINFO_A
+struct GetAddrInfoAsyncState
+{
+    struct gaicb gai_request;
+    struct gaicb* gai_requests;
+    struct sigevent sigevent;
+
+    struct addrinfo hint;
+    HostEntry* entry;
+    GetHostEntryForNameCallback callback;
+    char address[];
+};
+
+static void GetHostEntryForNameAsyncComplete(sigval_t context)
+{
+    struct GetAddrInfoAsyncState* state = (struct GetAddrInfoAsyncState*)context.sival_ptr;
+
+    atomic_thread_fence(memory_order_acquire);
+
+    GetHostEntryForNameCallback callback = state->callback;
+
+    int ret = ConvertGetAddrInfoAndGetNameInfoErrorsToPal(gai_error(&state->gai_request));
+
+    if (ret == 0)
+    {
+        const uint8_t* address = (const uint8_t*)state->address;
+        struct addrinfo* info = state->gai_request.ar_result;
+
+        ret = GetHostEntries(address, info, state->entry);
+    }
+
+    assert(callback != NULL);
+    callback(state->entry, ret);
+
+    free(state);
+}
+#endif
+
+int32_t SystemNative_PlatformSupportsGetAddrInfoAsync()
+{
+    return HAVE_GETADDRINFO_A;
+}
+
+int32_t SystemNative_GetHostEntryForName(const uint8_t* address, int32_t addressFamily, HostEntry* entry)
+{
+    if (address == NULL || entry == NULL)
+    {
+        return GetAddrInfoErrorFlags_EAI_BADARG;
+    }
+
+    sa_family_t platformFamily;
+    if (!TryConvertAddressFamilyPalToPlatform(addressFamily, &platformFamily))
+    {
+        return GetAddrInfoErrorFlags_EAI_FAMILY;
+    }
+
+    struct addrinfo hint;
+    memset(&hint, 0, sizeof(struct addrinfo));
```

Comment:
I'm not sure if this zeroing is necessary. Tests will succeed either way, in the docs I can't find clear hints.
Being in doubt, it's safer to zero it.

Same in L606.

Reply:
Any I'll refactor this into one function, as this is done in the async code-path too.